Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -1183,77 +1228,121 @@ WarpXOpenPMDPlot::WriteOpenPMDFieldsAll ( //const std::string& filename,
+                }
+            }
+        } // icomp setup loop
 
-    int const ncomp = mf[i].nComp();
-    for ( int icomp=0; icomp<ncomp; icomp++ ) {
-          std::string const & varname = varnames[icomp];
+        for ( int icomp=0; icomp<ncomp; icomp++ ) {
+            std::string const & varname = varnames[icomp];
 
-          // assume fields are scalar unless they match the following match of known vector fields
-          std::string field_name = varname;
-          std::string comp_name = openPMD::MeshRecordComponent::SCALAR;
-          GetMeshCompNames( i, varname, field_name, comp_name );
+            auto [field_str_end_index, mode_index] = GetFieldModeIndices(varname);
+            bool var_in_theta_mode = field_str_end_index < varname.size();
 
-          auto mesh = meshes[field_name];
-          auto mesh_comp = mesh[comp_name];
-          if ( first_write_to_iteration )
-          {
-             SetupMeshComp( mesh, full_geom, mesh_comp );
-             detail::setOpenPMDUnit( mesh, field_name );
+            std::string field_name = varname.substr(0,field_str_end_index);
+            std::string varname_no_mode = varname.substr(0,field_str_end_index);
+            std::string comp_name = openPMD::MeshRecordComponent::SCALAR;
+            // assume fields are scalar unless they match the following match of known vector fields
+            GetMeshCompNames( i, varname_no_mode, field_name, comp_name );
 
-             auto relative_cell_pos = utils::getRelativeCellPosition(mf[i]);     // AMReX Fortran index order
-             std::reverse( relative_cell_pos.begin(), relative_cell_pos.end() ); // now in C order
-             mesh_comp.setPosition( relative_cell_pos );
-           }
+            auto mesh = meshes[field_name];
+            auto mesh_comp = mesh[comp_name];
 
-           // Loop through the multifab, and store each box as a chunk,
-           // in the openPMD file.
-           for( amrex::MFIter mfi(mf[i]); mfi.isValid(); ++mfi )
-           {
+            // Loop through the multifab, and store each box as a chunk,
+            // in the openPMD file.
+            for( amrex::MFIter mfi(mf[i]); mfi.isValid(); ++mfi )
+            {
                 amrex::FArrayBox const& fab = mf[i][mfi];
                 amrex::Box const& local_box = fab.box();
 
                 // Determine the offset and size of this chunk
                 amrex::IntVect const box_offset = local_box.smallEnd() - global_box.smallEnd();
-                auto const chunk_offset = getReversedVec( box_offset );
-                auto const chunk_size = getReversedVec( local_box.size() );
+                auto chunk_offset = getReversedVec( box_offset );
+                auto chunk_size = getReversedVec( local_box.size() );
+
+
+#if defined(WARPX_DIM_RZ)
+                if (var_in_theta_mode) {
+                    chunk_offset.emplace(chunk_offset.begin(), mode_index);
+                    chunk_size.emplace(chunk_size.begin(), 1);
```

Comment:
See discussion above :)
```suggestion
                    chunk_offset.insert(chunk_offset.begin(), mode_index);
                    chunk_size.insert(chunk_size.begin(), 1);
```